### PR TITLE
Document check-config.h and *adjust*.h as internal headers

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
It's technically possible to `#include` those headers, so users are doing it, and then complaining about the consequences. Resolve https://github.com/Mbed-TLS/mbedtls/issues/9147.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** no, but actually yes
- [x] **3.6 backport** https://github.com/Mbed-TLS/mbedtls/pull/9146
- [x] **2.28 backport** N/A
- [x] **tests** no (documentation only)

### CODE REVIEW

**Review by minosgalanakis — DISMISSED** [review]
LGTM

**Review by mpg — APPROVED** [review]
LGTM

File: include/mbedtls/config_adjust_legacy_crypto.h [hunk 1]
```diff
@@ -2,6 +2,8 @@
  * \file mbedtls/config_adjust_legacy_crypto.h
  * \brief Adjust legacy configuration configuration
  *
+ * This is an internal header. Do not include it directly.
```
Comment (gilles-peskine-arm) [inline include/mbedtls/config_adjust_legacy_crypto.h:5]: Note: https://github.com/Mbed-TLS/mbedtls/pull/9057 adds a new adjust file. If it gets merged first, we need to ensure that it follows the template that we're changing here.

**Comment by gilles-peskine-arm** [comment]
Although 7f19db535507c9ab676b78505cafc33d8fe8982e has been approved, there is more work to do:

* Forward-port the added enforcement from https://github.com/Mbed-TLS/mbedtls/pull/9146.
* Rebase on top of `development` since #9057 has been merged and it adds a `config_adjust` header where I need to add enforcement.

File: ChangeLog.d/check-config.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Changes
+   * Explicitly state that mbedtls/check_config.h must not be included manually.
```
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply ChangeLog.d/check-config.txt:2]: I'll forward-port once the 3.6 PR has been approved.

**Comment by gilles-peskine-arm** [comment]
I've rebased to cover the new header file added in https://github.com/Mbed-TLS/mbedtls/pull/9057, and forward-ported the compile-time errors added in the 3.6 version. So this is now the proposed final version of this PR.

**Review by ronald-cron-arm — APPROVED** [review]

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Pull request merged**